ionoscloud/resource_kafka_cluster.go

```diff
@@ -0,0 +1,205 @@
+			},
+			"broker_addresses": {
+				Type:        schema.TypeList,
+				Description: "IP address and port of cluster brokers.",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceKafkaClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).KafkaClient
+
+	createdCluster, _, err := client.CreateCluster(ctx, d)
+	if err != nil {
+		d.SetId("")
+		diags := diag.FromErr(fmt.Errorf("error creating Kafka Cluster: %w", err))
+		return diags
+	}
+
+	d.SetId(*createdCluster.Id)
+	log.Printf("[INFO] Created Kafka Cluster: %s", d.Id())
+
+	// Sleep for 5 second to avoid 500 error from the API
+	time.Sleep(5 * time.Second)
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterAvailable)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("an error occurred while Kafka Cluster waiting to be ready: %w", err))
+		return diags
+	}
+
+	return resourceKafkaClusterRead(ctx, d, meta)
+}
+
+func resourceKafkaClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).KafkaClient
+
+	cluster, apiResponse, err := client.GetClusterByID(ctx, d.Id(), d.Get("location").(string))
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		diags := diag.FromErr(fmt.Errorf("error while fetching Kafka Cluster %s: %w", d.Id(), err))
+		return diags
+	}
+
+	log.Printf("[INFO] Successfully retreived Kafka Cluster %s: %+v", d.Id(), cluster)
+
+	if err := client.SetKafkaClusterData(d, &cluster); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceKafkaClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).KafkaClient
+
+	apiResponse, err := client.DeleteCluster(ctx, d.Id(), d.Get("location").(string))
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		diags := diag.FromErr(fmt.Errorf("error while deleting Kafka Cluster %s: %w", d.Id(), err))
+		return diags
+	}
+
+	err = utils.WaitForResourceToBeDeleted(ctx, d, client.IsClusterDeleted)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("the check for Kafka Cluster deletion failed with the following error: %w", err))
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceKafkaClusterImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), ":")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("expected ID in the format location:cluster_id")
+	}
+
+	if err := d.Set("location", parts[0]); err != nil {
+		return nil, fmt.Errorf("failed to set location Kafka Cluster for import: %w", err)
+	}
+	d.SetId(parts[1])
+
+	diags := resourceKafkaClusterRead(ctx, d, meta)
+	if diags != nil && diags.HasError() {
```

Comment:
you can keep only the second check.

Reply:
that's cool, but I think I would like to keep it for clarity's sake